### ISSUE TITLE
control number of instances on each broker

### ISSUE THREAD

**Issue by cesar2064** [moleculerjs/moleculer#237]
Hi guys is there a way to control the number of instances on each broker?

**Comment by icebob** [comment]
The Broker is running your services. It can run one or more services. And you can execute multiple brokers too. So it's your choice that how many broker instances are running.

**Comment by cesar2064** [comment]
Yes, I understand it, but does each service run in a separate node js instance?. What happens if I want to use a service which is heavy and I want it to use more cores?, like PM2 does or node js cluster mode.

**Comment by tinchoz49** [comment]
HI @cesar2064 how are you?

You can have multiple services working in one node js instance, it's just business logic. 

If you want to have multiple node js instances you have many options: 
1. clustering using moleculer-runner with the flag `--instances`
2. using --scale flag if you are using docker
3. kubernetes
4. or just executing multiple times your node js process

Remember that for all of this cases you have to define a transport in order to allow to work the service discovery of moleculer.

This is something really cool to understand the different possible architectures for your nodes and services: http://moleculer.services/0.12/docs/nodes.html

**Comment by cesar2064** [comment]
this is so cool, thanks @tinchoz49 and @icebob
